test/TestUtilities/Test.Utility/MockServer.cs

```diff
@@ -69,7 +66,37 @@ public MockServer()
         /// </summary>
         public void Start()
         {
-            Listener.Start();
+            // Try creating and starting
+
+            int attempts = 1;
+            do
+            {
+                try
+                {
+                    // tests that cancel downloads and exit will cause the mock server to throw, this should be ignored.
+                    _listener = new HttpListener()
+                    {
+                        IgnoreWriteExceptions = true
+                    };
+
+                    _listener.Prefixes.Add(PortReserver.BaseUri);
+                    _listener.AuthenticationSchemes = _authenticationSchemes;
+                    _listener.Start();
+                }
+                catch (Exception)
+                {
+                    _listener = null;
+
+                    if (attempts++ >= 5)
+                    {
+                        throw;
+                    }
+
+                    Thread.Sleep(50);
```

Comment:
This doesn't seem enough of a delay before a retry. I would do 200ms which would be a total of 1 sec with 5 retries.